src/main/java/com/amazon/opendistroforelasticsearch/sql/request/SqlRequest.java

```diff
@@ -36,11 +36,22 @@
 
     String sql;
     JSONObject jsonContent;
+    String cursor;
```

Comment:
Since we're adding new fields to the protocol and version information is missing, could you add a `version` field too? For example, `version` missing or 1.0 means "old" client code and `version` = 2.0 means "new".

Reply:
I took a quick look into version field of other protocol. Take Internet Protocol (IP) for example, the first field in the header is version which indicates IPv4 or IPv6. What follows this field in the header differs for different version so application is aware of how to parse the entire header.

In our case, because we are only adding new fields, probably we can delay adding version field. We need to reconsider this and may need to refactor our protocol carefully (ex. divide into header and body) when adding such as binary protocol.

Reference: https://en.wikipedia.org/wiki/IPv4#Version and https://en.wikipedia.org/wiki/IPv6_packet#Fixed_header

Reply:
Yes, we can add this later if needed. This will not break existing clients now as we adding new field and not modifying/deleting existing one.